src/docbuilder/limits.rs

```diff
@@ -88,15 +87,102 @@ impl Limits {
     }
 }
 
-fn scale(mut value: usize, interval: usize, labels: &[&str]) -> String {
+const TIME_SCALE: fn(usize) -> String = |v| scale(v, 60, &["seconds", "minutes", "hours"]);
+const SIZE_SCALE: fn(usize) -> String = |v| scale(v, 1024, &["bytes", "KB", "MB", "GB"]);
+
+fn scale(value: usize, interval: usize, labels: &[&str]) -> String {
+    let (mut value, interval) = (value as f64, interval as f64);
     let mut chosen_label = &labels[0];
     for label in &labels[1..] {
-        if (value as f64) / (interval as f64) >= 1.0 {
+        if value / interval >= 1.0 {
             chosen_label = label;
             value = value / interval;
         } else {
             break;
         }
     }
+    // 2.x
+    let mut value = format!("{:.1}", value);
+    // 2.0 -> 2
+    if &value[value.len() - 1..] == "0" {
```

Comment:
This can be simplified with [`ends_with`](https://doc.rust-lang.org/std/primitive.str.html#method.ends_with) 🙂 

```suggestion
    if value.ends_with(".0") {
```